test/asmcup/runtime/RobotTest.java

```diff
@@ -0,0 +1,52 @@
+package asmcup.runtime;
+
+import asmcup.vm.VM;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+public class RobotTest {
+	Robot robot = null;
+
+	@Before
+	public void setUp() {
+		robot = new Robot(42);
+		robot.position(42, 21);
+	}
+
+	@Test
+	public void testAccelerometer() {
+		World world = new World();
+		// Read Accelerometer
+		VM vm = robot.getVM();
+		vm.push8(Robot.IO_ACCELEROMETER);
+		vm.setIO(true);
+		robot.handleIO(world);
+
+		// We did not drive, so we should expect 0 values.
+		float x = vm.popFloat();
+		float y = vm.popFloat();
+		assertTrue("X is 0", Math.abs(0.0f - x) < 0.001f);
+		assertTrue("Y is 0", Math.abs(0.0f - y) < 0.001f);
+		assertTrue("X == Y", Math.abs(x - y) < 0.001f);
+
+		// Drive a little bit
+		robot.setMotor(1.0f);
+		robot.setSteer(0.5f);
+		robot.tickHardware(world);
+
+
+		vm.push8(Robot.IO_ACCELEROMETER);
+		vm.setIO(true);
+		robot.handleIO(world);
+
+		// We drove, so we should not expect 0 values.
+		x = vm.popFloat();
+		y = vm.popFloat();
+		assertTrue("X is 0", Math.abs(0.0f - x) > 0.001f);
```

Comment:
What if the bot starts on a solid tile? I could get behind placing the bot in the middle of an "open" tile, as that would guarantee it has a distance to drive until it rams a wall (because it's smaller than a tile).